addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -149,9 +154,9 @@ public LEDStateDTO withIncrementedProgramSpeed(int step) {
     }
 
     public Color getColor() {
-        float hue = (float) this.getHue().intValue() / 360;
-        float saturation = (float) this.getSaturation().doubleValue() / 100;
-        float brightness = (float) this.getBrightness().intValue() / 100;
+        float hue = (float) this.getHue().doubleValue() / 360.0f;
```

Comment:
_will it allow us to get rid of the or some of the other calculations as well?_
It will not git rid of the other calculations.

I tried to use : HSBType fromRGB(int r, int g, int b) { ..} but fromRGB will also cast to integers. see : https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/HSBType.java#L114 
I think this is a bug but would like to get educated of the reason why casting to int is done here.

example:
```
		HSBType test = HSBType.fromRGB(128,4,200);
		System.out.println((test.getRGB()>>16 & 0xff) + " " + (test.getRGB()>>8 & 0xff) + " " + (test.getRGB() & 0xff));
```

Will output 124 3 198 instead of 128,4,200